src/jllvm/compiler/ClassObjectStubCodeGenerator.cpp

```diff
@@ -146,16 +188,25 @@ llvm::Function* jllvm::generateFieldAccessStub(llvm::Module& module, const Class
     llvm::IRBuilder<> builder(llvm::BasicBlock::Create(module.getContext(), "entry", function));
 
     // Static field accesses trigger class object initializations.
-    if (field->isStatic() && !classObject.isInitialized())
+    if (field->isStatic() && !classObject.markedInitialized())
     {
         buildClassInitializerInitStub(builder, classObject);
     }
 
     llvm::Value* returnValue;
     if (field->isStatic())
     {
-        returnValue = builder.CreateIntToPtr(
-            builder.getInt64(reinterpret_cast<std::uint64_t>(field->getAddressOfStatic())), returnType);
+        // Only sound if field is static, final, initialized and not a reference type
+        if (classObject.isInitialized() && !descriptor.isReference())
```

Comment:
There's a check missing here that the field is final which is causing the test failures I think.

Shows that the transformation itself works at least :) although a little too well

Reply:
I had it and removed it by mistake